### PR TITLE
added roxygen comment template, introduced .autoDoc() function, updated vignette

### PR DESCRIPTION
**Pull request by paulobrecht**
I added a script called autoDoc.R to inst/extdata/ and changed the generation of datasets.R from a writelines() call to a script in inst/extdata/. I updated the datapackage.skeleton() function to copy those two files into the data-raw directory of the data package skeleton being created.

After running datapackage.skeleton(), the user adds their raw data to inst/extdata as usual. They then update datasets.R to call whatever script(s) they want and they define objectsToKeep, which is the argument to keepDataObjects().

However, objectsToKeep also becomes an input to the new autoDoc.R script. It generates boilerplate roxygen comments for the package as well as each object listed in objectsToKeep. For each object in objectsToKeep, the script generates data set roxygen documentation, include the tedious-to-do-manually set of \item{varname}{} lines in \describe. For each one, it also generates the \seealso lines for every object except this object itself.

The first time the user's package is built, this boilerplate is written to a file called edit_and_rename_to_'documentation.R'.R. This file is sourced by datasets.R on the first run and becomes the roxygen documentation and thus the .Rd help files. However, as suggested by the name, when the user edits the file and adds variable descriptions and detailed package-specific info and renames it to documentation.R, then on subsequent package builds the roxygen comments are sourced from this edited documentation.R file instead.

I think separating out the roxygen comments from datasets.R into documentation.R helps keep things modular by separating the processing code from the help file generation, using objectsToKeep as a way of tracking on the fly which objects need to be documented.

### CODE REVIEW

File: R/01.R [hunk 1]
```diff
@@ -139,6 +139,68 @@
...
 }
 
+
+# function .rc() prepends "#'" to a string or character vector
+.rc <- function(strVec){paste("#'", strVec)}
+
+# function .autoDoc() automates the creation of a basic roxygen template for the package and each object in objectsToKeep
+# arguments are pname and ds2kp, normally defined in datasets.R
+# pname is name of package, ds2kp is list of objects to save in data package
+.autoDoc <- function(pname, ds2kp, env){
```
Comment (paulobrecht) [inline R/01.R]: dataEnv and the .autodoc() function environment are both children of .GlobalEnv. Inside the function, topenv() is the .autoDoc() function environment and dataEnv is invisible. My best solution (workaround?) is to pass dataEnv as an argument into the function. From .GlobalEnv, topenv() correctly points to dataEnv when calling the function.

File: inst/extdata/datasets.R [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# ------------------------------------------------------------
+# Auto build roxygen documentation
+# On first build, we generate boilerplate roxygen documentation using preprocessData:::.autoDoc()
+# User then manually edits the output file edit_and_rename_to_'documentation.R'.R and renames it to documentation.R.
+# The documentation.R file is then used for all subsequent builds.
+if(file.exists("documentation.R")){
+  sys.source('documentation.R', envir=topenv())
+} else {
+  preprocessData:::.autoDoc(pkgName, objectsToKeep, topenv())
```
Comment (paulobrecht) [inline inst/extdata/datasets.R]: Passing topenv() in as an argument is my solution to the problem mentioned in the other line note.

**Comment by paulobrecht** [comment]
I made the changes we discussed. There's a new function .autoDoc(), and I read in the datasets.R template using system.file(). I also updated the vignette and its rendered HTML doc, but did not update the README.md file.

File: inst/extdata/datasets.R (hunk 2 above)
Comment (gfinak) [inline inst/extdata/datasets.R]: Yep, I think that's fine.

**Pull request merged**